tests/chainer_tests/links_tests/loss_tests/test_hierarchical_softmax.py

```diff
@@ -107,5 +125,52 @@ def test_to_cpu(self):
         self.assertTrue((f.paths == g.paths).all())
         self.assertTrue((f.codes == g.codes).all())
 
+    def compute_argmax(self, x, gpu):
```

Comment:
It has only five leaves. You don't need to write general code. Instead manually calculates their scores.
```
def node_score(self, node_index, x, code):
  return 1 / (numpy.exp(self.W[node_index].dot(x) * code) + 1)
# (0: (1: 0, 1), (2: (3: 2, 3), 4))
scores[0] = self.node_score(0, x, 1) * self.node_score(1, x, 1)
scores[1] = self.node_score(0, x, 1) * self.node_score(1, x, -1)
...
```

Reply:
Thank you for your comments!
I am fixing my code, node_score function should be following code?
```
def node_score(self, node_index, x, code):
  return 1 / (numpy.exp(- self.W[node_index].dot(x) * code) + 1)
```
I think sigmoid function needs ` 1 / exp(-x)` ?